pkg/common-controller/snapshot_controller.go

```diff
@@ -812,22 +812,24 @@ func (ctrl *csiSnapshotCommonController) ensurePVCFinalizer(snapshot *crdv1.Volu
 		return newControllerUpdateError(snapshot.Name, "cannot get claim from snapshot")
 	}
 
-	if pvc.ObjectMeta.DeletionTimestamp != nil {
-		klog.Errorf("cannot add finalizer on claim [%s] for snapshot [%s]: claim is being deleted", pvc.Name, snapshot.Name)
-		return newControllerUpdateError(pvc.Name, "cannot add finalizer on claim because it is being deleted")
+	if utils.ContainsString(pvc.ObjectMeta.Finalizers, utils.PVCFinalizer) {
+		klog.Infof("Protection finalizer already exists for persistent volume claim %s/%s", pvc.Name, pvc.Namespace)
+		return nil
 	}
 
-	// If PVC is not being deleted and PVCFinalizer is not added yet, the PVCFinalizer should be added.
-	if pvc.ObjectMeta.DeletionTimestamp == nil && !utils.ContainsString(pvc.ObjectMeta.Finalizers, utils.PVCFinalizer) {
-		// Add the finalizer
+	if pvc.ObjectMeta.DeletionTimestamp != nil {
+		klog.Errorf("cannot add finalizer on claim [%s/%s] for snapshot [%s/%s]: claim is being deleted", pvc.Name, pvc.Namespace, snapshot.Name, snapshot.Namespace)
```

Comment:
Can you switch the order of name and namespace?
Print namespace/name for both pvc and snapshot